### ISSUE TITLE
qemu: uncaught target signal 4 (Illegal instruction) - core dumped

### ISSUE THREAD

**Issue by cowhca** [Dullage/eyelign#21]
I have docker running and run 
`docker run -it --rm \
  -v "INPUT_DIR" \
  -v "OUTPUT_DIR" \
  dullage/eyelign:latest`

It tells me this and hangs:
`qemu: uncaught target signal 4 (Illegal instruction) - core dumped`

I am running the command in a directory with INPUT_DIR and OUTPUT_DIR exist and INPUT_DIR has all the images as jpg files.

**Comment by cowhca** [comment]
It also gives me 
`WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested` 
but that goes away with `--platform linux/amd64` flag

**Comment by dullage** [comment]
Hi @cowhca. Are you running this on a machine with an arm processor. Perhaps a newer Mac?

If so, that's the problem as the Docker image I published was only built for x86.

**Comment by cowhca** [comment]
Yes, I'm running it on an M1 MacBook Pro. Thanks for letting me know, unfortunate it won't work for me.

**Comment by dullage** [comment]
When I get a chance, I'll see if I can do an arm build. It may be that some of the dependencies don't support it but I'll give it a shot and let you know.

**Comment by dullage** [comment]
@cowhca - There is now a build for arm/v8 available on dockerhub. Hopefully, the example docker run command from the README page should work for you now.

**Comment by duceduc** [comment]
Hello. I am experiencing similar issue. Trying this on a rpi4.

    ngx@ngx:/tmp $ docker pull dullage/eyelign:latest
    latest: Pulling from dullage/eyelign
    no matching manifest for linux/arm64/v8 in the manifest list entries

**update**: I have ran the below with the --platform tag mentioned above. It installed, but error when trying to run it.

    ngx@ngx:/tmp $ docker pull --platform linux/amd64 dullage/eyelign:latest

Error:

    ngx@ngx:/tmp $ docker run -it --rm \
      -v "/var/www/log/shots:/input" \
      -v "/var/www/log/aligned:/ouput" \
      dullage/eyelign:latest
    WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
    exec /usr/local/bin/python3: exec format error

**Comment by dullage** [comment]
@duceduc - I haven't tried running it on a Pi4 but I did just try pulling the image on my Pi4 and it worked. 

Can you run the following command and see what's in the output: `docker manifest inspect dullage/eyelign:latest`.

**Comment by duceduc** [comment]
Hi. This is the output.

    ngx@ngx:/tmp $ docker manifest inspect dullage/eyelign:latest
    {
       "schemaVersion": 2,
       "mediaType": "application/vnd.docker.distribution.manifest.list.v2+json",
       "manifests": [
          {
             "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
             "size": 2843,
             "digest": "sha256:e85474190cb45016746d1fab2ba973c4f572fe9960e7454d23e02831de31a8ed",
             "platform": {
                "architecture": "amd64",
                "os": "linux"
             }
          },
          {
             "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
             "size": 2843,
             "digest": "sha256:75368129361bb4a4bd268535efbd445fbc4e1f37e89c27c50906f1ef05a3da33",
             "platform": {
                "architecture": "arm",
                "os": "linux",
                "variant": "v8"
             }
          }
       ]
    }

My rpi4

    ngx@ngx:/tmp $ cat /sys/firmware/devicetree/base/model;echo
    Raspberry Pi 4 Model B Rev 1.4

**Comment by dullage** [comment]
Thanks @duceduc. I've pushed a new build-up that uses slightly different architecture tags. Can you give a pull another try please.

**Comment by duceduc** [comment]
Hi. Thank you for for the update. It is working now. Now I just need to play with the options. Having trouble finding the eyes.
